cmd/argocd-repo-server/commands/argocd_repo_server.go

```diff
@@ -189,6 +191,7 @@ func NewCommand() *cobra.Command {
 	command.Flags().StringVar(&otlpAddress, "otlp-address", env.StringFromEnv("ARGOCD_REPO_SERVER_OTLP_ADDRESS", ""), "OpenTelemetry collector address to send traces to")
 	command.Flags().BoolVar(&disableTLS, "disable-tls", env.ParseBoolFromEnv("ARGOCD_REPO_SERVER_DISABLE_TLS", false), "Disable TLS on the gRPC endpoint")
 	command.Flags().StringVar(&maxCombinedDirectoryManifestsSize, "max-combined-directory-manifests-size", env.StringFromEnv("ARGOCD_REPO_SERVER_MAX_COMBINED_DIRECTORY_MANIFESTS_SIZE", "10M"), "Max combined size of manifest files in a directory-type Application")
+	command.Flags().StringArrayVar(&cmpTarExcludedGlobs, "cmp-tar-exclude", env.StringsFromEnv("ARGOCD_REPO_SERVER_CMP_TAR_EXCLUSIONS", []string{".git"}, ";"), "Globs to filter when sending tarballs to sidecar CMPs.")
```

Comment:
Can we just make it an empty string by default? I think @leoluz was right when he first wrote this code, we should make exclusions opt-in to avoid unpleasantly surprising the user.

Reply:
That would be much preferred! Will do.